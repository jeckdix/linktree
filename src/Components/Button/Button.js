import React from "react";
import styled from "styled-components";
import { Link } from "react-router-dom";


const StyledLink = styled(Link)`
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  padding: 24px 32px;
  gap: 8px;
  background: #D0D5DD;
  border: 1px solid #D0D5DD;
  border-radius: 8px;
  text-decoration: none;
  width: 355px;
  height: 68px;
  :hover {
    background: #EAECF0; 
  } 
  @media (min-width: 1200px) {
    width: 1152px;
    height: 76px;
  }
`;

const StyledTitle = styled.h3`
  font-family: "Inter";
  font-style: normal;
  font-weight: 500;
  font-size: 14px;
  line-height: 20px;
  display: flex;
  align-items: center;
  text-align: center;
  color: #101828;
  flex: none;
  order: 0;
  flex-grow: 0;


  @media (min-width: 1200px) {
    font-weight: 600px;
    line-height: 28px;
  }
`;

const Button = (props) => {
  return (
      <StyledLink to={props.link} id={props.id}>
        <StyledTitle>{props.title}</StyledTitle>
      </StyledLink>
  );
};

export default Button;
  