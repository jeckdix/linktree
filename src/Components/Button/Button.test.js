import React from 'react'
import { render, screen, cleanup } from '@testing-library/react'
import '@testing-library/jest-dom'
import Button from './Button'


describe('Button', () => {
    afterAll(() => {
        cleanup();
    });

    it('should render without errors', () => {
        const linkProps = {
                id: "btn__zuri",
                link: "https://training.zuri.team/",
                title: "Zuri Team",
        }
        const { asFragment } = render(<Button {...linkProps} />);
        expect(asFragment()).toMatchSnapshot();
    });

    it('should contain the correct attributes', () => {
            const linkProps = {
                    id: "btn__zuri",
                    link: "https://training.zuri.team/",
                    title: "Zuri Team",
            }
            render(<Button {...linkProps} />);

            expect(screen.getByRole('link')).toHaveAttribute('href', linkProps.link);
            expect(screen.getByRole('link')).toHaveAttribute('id', linkProps.id);
            expect(screen.getByRole('link')).toHaveTextContent(linkProps.title);
    })
})

// jestdom
// react-testing-library



