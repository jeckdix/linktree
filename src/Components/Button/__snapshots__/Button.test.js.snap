// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render without errors 1`] = `
<DocumentFragment>
  <a
    class="sc-bcXHqe jAgPMp"
    href="https://training.zuri.team/"
    id="btn__zuri"
  >
    <h3
      class="sc-gswNZR fEFhHF"
    >
      Zuri Team
    </h3>
  </a>
</DocumentFragment>
`;
